src/pages/settings/ProfilePage.js

```diff
@@ -1,24 +1,236 @@
-import React from 'react';
+import React, {Component} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import {
+    View,
+    TextInput,
+} from 'react-native';
+import RNPickerSelect from 'react-native-picker-select';
+import Str from 'expensify-common/lib/str';
+import moment from 'moment-timezone';
+
 import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
 import Navigation from '../../libs/Navigation/Navigation';
 import ScreenWrapper from '../../components/ScreenWrapper';
+import {setPersonalDetails} from '../../libs/actions/PersonalDetails';
 import ROUTES from '../../ROUTES';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+import Avatar from '../../components/Avatar';
+import styles from '../../styles/styles';
+import Text from '../../components/Text';
+import {DownArrow} from '../../components/Icon/Expensicons';
+import Icon from '../../components/Icon';
+import Checkbox from '../../components/Checkbox';
+import ButtonWithLoader from '../../components/ButtonWithLoader';
+
+const propTypes = {
+    /* Onyx Props */
+    // The personal details of the person who is logged in
+    myPersonalDetails: PropTypes.shape({
+        // Email/Phone login of the current user from their personal details
+        login: PropTypes.string,
+
+        // Display first name of the current user from their personal details
+        firstName: PropTypes.string,
+
+        // Display last name of the current user from their personal details
+        lastName: PropTypes.string,
+
+        // Avatar URL of the current user from their personal details
+        avatarURL: PropTypes.string,
+
+        // Pronouns of the current user from their personal details
+        pronouns: PropTypes.string,
+
+        // timezone of the current user from their personal details
+        timezone: PropTypes.shape({
+
+            // Value of selected timezone
+            selected: PropTypes.string,
+
+            // Whether timezone is automatically set
+            automatic: PropTypes.bool,
+        }),
+    }),
+};
+
+const defaultProps = {
+    myPersonalDetails: {},
+};
+
+const pronounsMap = {
+    theyThemTheirs: 'They/them/theirs',
+    sheHerHers: 'She/her/hers',
+    heHimHis: 'He/him/his',
+    zeHirHirs: 'Ze/hir/hirs',
+    selfSelect: 'Self-select',
+    callMeByMyName: 'Call me by my name',
+};
+
+const timezones = moment.tz.names()
+    .map(timezone => ({
+        value: timezone,
+        label: timezone,
+    }));
+
+class ProfilePage extends Component {
+    constructor(props) {
+        super(props);
+
+        const {
+            firstName, lastName, pronouns, timezone = {},
+        } = props.myPersonalDetails;
+
+        this.state = {
+            firstName,
+            lastName,
+            pronouns,
+            selfSelectedPronouns: '',
```

Comment:
Have one thing left to do but need some direction. Basically if a user has a self-selected pronoun, it's not automatically setting the dropdown.

I can check if the pronouns exist in the map above and default to self-selected if not, but that might not be a future-proof approach if the list changes.

The other option would be if `isSelfSelected` or something like that was saved as part of the personalDetails to know whether the pronouns were self selected.  (this approach would require work outside of the scope of this ticket but imo the better route to go)

Let me know if any of this needs further clarification.

Reply:
> I can check if the pronouns exist in the map above and default to self-selected if not

This sounds fine to me. 

> that might not be a future-proof approach if the list changes

How do you mean?

Reply:
FWIW, I think if a user types in a personal pronoun that's an exact match to an item we already have in the list, then it's ok to just default to select that item from the list.

Reply:
sounds good, just wanted to make sure the inverse would also be fine. for example if I select "Call me by my name" and a new release no longer has that as a dropdown item then it would show up as "Self-Select" and  "Call me by my name"

Reply:
yup, this is fine too